numba/tests/test_ufuncs.py

```diff
@@ -1014,6 +1017,65 @@ def myadd(a0, a1):
             cr = compile_isolated(myadd, (arr_ty, arr_ty),
                                   flags=no_pyobj_flags)
 
+    def test_broadcast_implicit_output_npm_nrt(self):
+        def pyfunc(a0, a1):
+            return np.add(a0, a1)
+
+        input1_operands = [
+            np.arange(3, dtype='u8'),
+            np.arange(3*3, dtype='u8').reshape(3,3),
+            np.arange(3*3*3, dtype='u8').reshape(3,3,3),
+            np.arange(3, dtype='u8').reshape(3,1),
+            np.arange(3, dtype='u8').reshape(1,3),
+            np.arange(3, dtype='u8').reshape(3,1,1),
+            np.arange(3*3, dtype='u8').reshape(3,3,1),
+            np.arange(3*3, dtype='u8').reshape(3,1,3),
+            np.arange(3*3, dtype='u8').reshape(1,3,3)]
+
+        input2_operands = input1_operands
+
+        for x, y in itertools.product(input1_operands, input2_operands):
+
+            input1_type = types.Array(types.uint64, x.ndim, 'C')
+            input2_type = types.Array(types.uint64, y.ndim, 'C')
+
+            cr = self.cache.compile(pyfunc, (input1_type, input2_type),
+                                    flags=enable_nrt_flags)
+            cfunc = cr.entry_point
+
+            expected = np.add(x, y)
+            result = cfunc(x, y)
+            self.assertTrue(np.all(result == expected))
```

Comment:
It seems Numpy's `==` operator implementation doesn't check array shapes. Perhaps use `np.testing.assert_array_equal` instead?